descwl_coadd/coadd.py

```diff
@@ -1,1236 +1,639 @@
-        stats_ctrl.setAndMask(badmask)
-        stats_ctrl.setWeighted(True)
-
-        masked_images = [w.getMaskedImage() for w in wexps]
-        stacked_image = afw_math.statisticsStack(
-            masked_images, stats_flags, stats_ctrl, weights, 0, 0)
-
-        stacked_exp = afw_image.ExposureF(stacked_image, self.coadd_wcs)
-        # stacked_exp.setWcs(self.coadd_wcs)
-        stacked_exp.getInfo().setCoaddInputs(input_recorder.makeCoaddInputs())
-        coadd_inputs = stacked_exp.getInfo().getCoaddInputs()
-
-        # Build coadd psf
-        for wexp, weight in zip(wexps, weights):
-            input_recorder.addVisitToCoadd(coadd_inputs, wexp, weight)
-
-        if psf_config is not None:
-            coadd_psf = CoaddPsf(
-                coadd_inputs.ccds,
-                self.coadd_wcs,
-                psf_config.makeControl(),
-            )
-            stacked_exp.setPsf(coadd_psf)
+def extract_coadd_psf(coadd_psf_exp, logger):
+    """
+    extract the PSF image, zeroing the image where
+    there are "bad" pixels, associated with areas not
+    covered by the input psfs
 
-        return stacked_exp
+    Parameters
+    ----------
+    coadd_psf_exp: afw_image.ExposureF
+        The psf exposure
 
-    def _get_jac(self, *, cenx, ceny):
-        """
-        get jacobian at the coadd image center, and make
-        an ngmix jacobian with center specified (this is not the
-        position used to evaluate the jacobian)
-        """
-        import galsim
+    Returns
+    -------
+    KernelPsf
+    """
+    psf_image = coadd_psf_exp.image.array
 
-        crpix = self.galsim_wcs.crpix
-        galsim_pos = galsim.PositionD(x=crpix[0], y=crpix[1])
+    wbad = np.where(~np.isfinite(psf_image))
+    if wbad[0].size == psf_image.size:
+        raise ValueError('no good pixels in the psf')
 
-        galsim_jac = self.galsim_wcs.jacobian(image_pos=galsim_pos)
+    if wbad[0].size > 0:
+        logger.info('zeroing %d bad psf pixels' % wbad[0].size)
+        psf_image[wbad] = 0.0
 
-        return ngmix.Jacobian(
-            x=cenx,
-            y=ceny,
-            dudx=galsim_jac.dudx,
-            dudy=galsim_jac.dudy,
-            dvdx=galsim_jac.dvdx,
-            dvdy=galsim_jac.dvdy,
+    return KernelPsf(
+        FixedKernel(
+            afw_image.ImageD(psf_image.astype(np.float))
         )
+    )
 
-    def _get_psf_obs(self):
-        """
-        get the psf observation
-        """
-        crpix = self.galsim_wcs.crpix
-        stack_pos = geom.Point2D(crpix[0], crpix[1])
 
-        psf_obj = self.coadd_exp.getPsf()
-        psf_image = psf_obj.computeKernelImage(stack_pos).array
+def get_exp_and_noise(exp_or_ref, rng, remove_poisson):
+    """
+    get the exposure (possibly from a deferred handle) and create
+    a corresponding noise exposure
 
-        psf_cen = (np.array(psf_image.shape)-1.0)/2.0
+    TODO move interpolating BRIGHT into metadetect or mdet-lsst-sim
```

Comment:
We cannot move this to the sim code. It needs to be in mdet or in the stack task that runs mdet on the coadds. This last option is the equivalent of where it is in DES.

Reply:
mdet-lsst-sim isn't the sim code it is a code that runs metadetection on the sims

https://github.com/esheldon/mdet-lsst-sim

This is what I've used to run all the lsst tests

Reply:
Right yes. In any case, I suspect we'll need to move the interp to the stack task that runs mdet.

Reply:
We need to have a single code path for this so we can test it once.  I don't want it to be both in the stack task and in a separate code we run on sims.

I think this leaves only metadetect as the place for it